concepts/objects/about.md

```diff
@@ -0,0 +1,251 @@
+For keys with special characters in the name, you need to apply the usual string notation.
+
+```javascript
+const obj = {
+  '1keyStartsWithNumber': '...',
+  'key/with/slashes': '...',
+  'key-with-dashes': '...',
+  'key with spaces': '...',
+  '#&()[]{}èä樹keyWithSpecialChars': '...',
+};
+```
+
+Often times you want to combine existing variables into an object.
+This would often lead to key-value pairs like `name: name`.
+JavaScript has a short-hand notation that allows to just write `name` instead.
+There is also a short-hand notation for defining a function in an object.
+There you can omit the colon and the `function` keyword.
+
+```javascript
+const x = 1;
+const y = 2;
+
+const obj = {
+  x: x,
+  y: y,
+  calcSum: function (a, b) {
+    return a + b;
+  },
+};
+
+// Can be shortened to ...
+const obj = {
+  x,
+  y,
+  calcSum(a, b) {
+    return a + b;
+  },
+};
+```
+
+If you want to create an object literal with a key name that is stored in a variable or derived via an expression, you need to wrap the variable or expression in square brackets.
+
+```javascript
+const key = 'greeting';
+
+const obj = {
+  [key]: 'hello world',
+};
+```
+
+## Retrieving a Value
+
+There are two ways to retrieve the value for a given key, dot notation and bracket notation.
+
+```javascript
+const obj = { greeting: 'hello world' };
+
+obj.greeting;
+// => hello world
+
+obj['greeting'];
+// => hello world
+
+// Bracket notation also works with variables.
+const key = 'greeting';
+obj[key];
+// => hello world
+```
+
+Using the dot notation as a short-hand has the same restriction as omitting the quotation marks.
+It only works if the key follows the identifier naming rules.
+
+If you try to retrieve a key that does not exist in the object, JavaScript returns `undefined`.
+See the [Null and Undefined Concept][concept-null-undefined] for more details on this.
+
+You can chain the keys if you want to retrieve a value from a nested object.
+
+```javascript
+const obj = {
+  address: {
+    street: 'Trincomalee Highway',
+    city: 'Batticaloa',
+  },
+};
+
+obj.address.city;
+// => 'Batticaloa'
+
+obj['address']['city'];
+// => 'Batticaloa'
+```
+
+## Adding or Changing a Value
+
+You can add or change a value using the assignment operator `=`.
+Again, there is dot and bracket notation available.
+
+```javascript
+const obj = { greeting: 'hello world' };
+
+obj.greeting = 'Hi there!';
+obj['greeting'] = 'Welcome.';
+
+obj.newKey1 = 'new value 1';
+obj['new key 2'] = 'new value 2';
+
+const key = 'new key 3';
+obj[key] = 'new value 3';
+```
+
+## Deleting an Entry
+
+You can delete a key-value pair from an object using the `delete` keyword.
+
+```javascript
+const obj = {
+  key1: 'value1',
+  key2: 'value2',
+};
+
+delete obj.key1;
+delete obj['key2'];
+```
+
+Note that although `undefined` is returned for non-existing keys, setting a key to `undefined` does not remove the entry.
```

Comment:
Very good advice!

Might be worth showing `hasOwnProperty` to validate something is gone, given that you show it below?

Reply:
I think just showing it below is enough here.